executor_manager/routers/routers.py

```
@@ -895,6 +913,323 @@ async def cancel_task(request: CancelTaskRequest, http_request: Request):
+                    f"[v1/cancel] Successfully cancelled task {request.task_id} "
+                    f"in container {request.executor_name}"
+                )
+
+                # Clean up heartbeat data
+                await _cleanup_task_heartbeat(request.task_id)
+
+                return {
+                    "status": "success",
+                    "message": f"Task {request.task_id} cancellation requested",
+                    "executor_name": request.executor_name,
+                }
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)